packages/framework/src/Framework/HydeServiceProvider.php

```diff
@@ -32,11 +32,11 @@ public function register(): void
         $this->kernel->setSourceRoot(config('hyde.source_root', ''));
 
         $this->registerSourceDirectories([
-            HtmlPage::class => '_pages',
-            BladePage::class => '_pages',
-            MarkdownPage::class => '_pages',
-            MarkdownPost::class => '_posts',
-            DocumentationPage::class => '_docs',
+            HtmlPage::class => config('hyde.source_directories.'.HtmlPage::class, '_pages'),
+            BladePage::class => config('hyde.source_directories.'.BladePage::class, '_pages'),
+            MarkdownPage::class => config('hyde.source_directories.'.MarkdownPage::class, '_pages'),
+            MarkdownPost::class => config('hyde.source_directories.'.MarkdownPost::class, '_posts'),
+            DocumentationPage::class => config('hyde.source_directories.'.DocumentationPage::class, '_docs'),
```

Comment:
Could be improved by having a location service that could also find kebab-case class names in config assuming they resolve to a class in the main pages namespace

```suggestion
            HtmlPage::class => LocationService::getClassInfoFromConfig(class: HtmlPage::class, option: 'source_directories', default_ '_pages'),
            BladePage::class => LocationService::getClassInfoFromConfig(class: BladePage::class, option: 'source_directories', default_ '_pages'),
            MarkdownPage::class => LocationService::getClassInfoFromConfig(class: MarkdownPage::class, option: 'source_directories', default_ '_pages'),
            MarkdownPost::class => LocationService::getClassInfoFromConfig(class: MarkdownPost::class, option: 'source_directories', default_ '_posts'),
            DocumentationPage::class => LocationService::getClassInfoFromConfig(class: DocumentationPage::class, option: 'source_directories', default_ '_docs'),
```

Reply:
Moved to issue https://github.com/hydephp/develop/issues/1009